K2Bridge/Visitors/RangeVisitor.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Globalization;
+    using K2Bridge.Models.Request.Aggregations;
+    using K2Bridge.Models.Response;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(RangeAggregation rangeAggregation)
+        {
+            Ensure.IsNotNull(rangeAggregation, nameof(TermsAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            // KustoQL case expression
+            // case(
+            var caseExpression = $"{KustoQLOperators.Case}(";
+
+            // case() predicates
+            // foo > 0 and foo < 800, "0-800"
+            foreach (var range in rangeAggregation.Ranges)
+            {
+                var rangeClause = new Models.Request.Queries.RangeClause()
+                {
+                    FieldName = rangeAggregation.Field,
+                    GTEValue = range.From?.ToString() ?? "*",
+                    LTValue = range.To?.ToString() ?? "*",
+                };
+                rangeClause.Accept(this);
+
+                var bucketName = $"{range.From}-{range.To}";
```

Comment:
logical question, 
what happens if you have 2 ranges
1-100
50-90
?

Reply:
I guess we should stick to the behavior of Elastic. see what happens.

Reply:
After testing these cases (overlapping range), it turns out that the `case()` based Kusto query does not cover these. I am working on implementing the `union()` query that Asaf mentioned (see update in the Wiki page) at the beginning and it is working for these cases.

Because the query is more complex, it requires a bunch of changes to the Visitors (but Request/Response are unchanged).

Would you like me to add more commits to this already large PR, or should we merge it and then I can open another one today with the query changes. LMKWYT! @AsafMah @eladiw

Reply:
By popular vote, I will be pushing more commits :-)